app/uk/gov/hmrc/transitmovementsrouter/controllers/MessagesController.scala

```diff
@@ -177,30 +173,64 @@ class MessagesController @Inject() (
-    persistenceConnector
-      .patchMessageStatus(movementId, messageId, MessageUpdate(messageStatus))
-      .asPresentation
+    for {
+      persistenceResponse <- persistenceConnector
+        .patchMessageStatus(movementId, messageId, MessageUpdate(messageStatus))
+        .asPresentation
+      _ = pushNotificationsConnector
+        .postJSON(
+          movementId,
+          messageId,
+          jsonValue
+        )
+    } yield persistenceResponse
 
   def handleSdesResponse() =
     Action.async(parse.json) {
       implicit request =>
         implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromRequest(request)
-        (for {
-          sdesResponse <- parseAndLogSdesResponse(request.body)
-          (movementId, messageId) = extractMovementMessageId(sdesResponse)
-          persistenceResponse <- sdesResponse.notification match {
-            case SdesNotificationType.FileProcessed =>
-              updateStatus(movementId, messageId, MessageStatus.Success)
-            case SdesNotificationType.FileProcessingFailure =>
-              updateStatus(movementId, messageId, MessageStatus.Failed)
-            case _ => EitherT.rightT[Future, PresentationError]((): Unit)
-          }
-        } yield persistenceResponse).fold[Result](
-          error => Status(error.code.statusCode)(Json.toJson(error)),
-          _ => Ok
-        )
+        parseAndLogSdesResponse(request.body) match {
+          case Left(presentationError) =>
+            Future.successful(Status(presentationError.code.statusCode)(Json.toJson(presentationError)))
+          case Right(sdesResponse) =>
+            val (movementId, messageId) = extractMovementMessageId(sdesResponse)
+            (for {
+              persistenceResponse <- sdesResponse.notification match {
+                case SdesNotificationType.FileProcessed =>
+                  updateAndSendNotification(
+                    movementId,
+                    messageId,
+                    MessageStatus.Success,
+                    Json.toJson(
+                      Json.obj(
+                        "code" -> OK,
+                        "message" ->
+                          s"The message $messageId for movement $movementId was successfully processed"
+                      )
+                    )
+                  )
+                case SdesNotificationType.FileProcessingFailure =>
+                  updateAndSendNotification(
+                    movementId,
+                    messageId,
+                    MessageStatus.Failed,
+                    Json.toJson(
+                      PresentationError.internalServiceError()
+                    )
+                  )
+                case _ => EitherT.rightT[Future, PresentationError]((): Unit)
+              }
+            } yield persistenceResponse).fold[Result](
+              presentationError => {
+                pushNotificationsConnector.postJSON(movementId, messageId, Json.toJson(presentationError))
```

Comment:
If this is a bad request, it's not appropriate to send that to the trader because that's not a them problem, but an us problem.

This **must** be an internal server error.